src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -76,7 +77,7 @@
      * @return command object based on user input
      * @throws UnknownCommandException if valid command cannot be parsed from user input
      */
-    public Command parse(String input) throws UnknownCommandException {
+    public Command parse(String input) throws UnknownCommandException, CommandException {
```

Comment:
Would probably be good to generalize both the exceptions to something like a ParserException instead of re-using a CommandException since it is thrown by Parser class.